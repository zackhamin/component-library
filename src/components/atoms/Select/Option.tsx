interface OptionProps {
	children: React.ReactNode;
	value: string | number;
	isDisabled?: boolean;
}

const Option = ({ children, value, isDisabled }: OptionProps) => {
	return (
		<option value={value} disabled={isDisabled}>
			{children}
		</option>
	);
};

export { Option };
