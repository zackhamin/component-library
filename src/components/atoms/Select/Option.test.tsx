import { Option } from './Option';
import { render, screen } from '@testing-library/react';

const defaultProps = {
	value: '',
	children: 'Foo',
};

describe('Option', () => {
	it('renders correctly', () => {
		const { container } = render(<Option {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});

	it('should be disabled when disabled prop set', () => {
		render(<Option {...defaultProps} isDisabled />);
		expect(screen.getByRole('option')).toHaveAttribute('disabled');
	});
});
