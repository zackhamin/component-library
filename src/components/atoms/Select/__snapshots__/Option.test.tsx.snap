// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option renders correctly 1`] = `
<div>
  <option
    value=""
  >
    Foo
  </option>
</div>
`;
